test/Core.Test/Utilities/CoreHelpersTests.cs

```diff
@@ -20,20 +20,37 @@ public void GenerateComb_Success()
             // the comb are working properly
         }
 
-        // TODO: Probably make this a Theory with some more possibilties.
-        [Fact]
-        public void Batch_Success()
+        [Theory]
+        [InlineData(2, 5, new[] { 1, 2, 3, 4, 5, 6, 7, 8 , 9, 0 })]
+        [InlineData(2, 3, new[] { 1, 2, 3, 4, 5 })]
+        [InlineData(2, 1, new[] { 1, 2 })]
+        [InlineData(1, 1, new[] { 1 })]
+        [InlineData(2, 2, new[] { 1, 2, 3 })]
+        public void Batch_Success(int batchSize, int totalBatches, int[] collection)
         {
             // Arrange
-            var source = new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+            var hasRemainder = collection.Length % batchSize != 0;
```

Comment:
Sorry to needle here, but this can be done with smaller branches.

```
var remainder = collection.Length % batchSize;

var batches = collection.Batch(batchSize);

Assert.Equal(totalBatches, batches.Count());

foreach(var batch in batches.Take(totalBatches - 1))
{
    Assert.Equal(batchSize, batch.Count());
}

Assert.Equal(batches.Last(), remainder == 0 ? batchSize : remainder)
```

Reply:
Perfect, I will make that change.